saleor/checkout/tasks.py

```diff
@@ -109,3 +116,57 @@ def delete_expired_checkouts(
         else:
             task_logger.warning("Invocation limit reached, aborting task")
     return total_deleted, has_more
+
+
+@app.task
+def automatic_checkout_completion_task(
+    checkout_pk,
+    user_id,
+    app_id,
+):
+    """Try to automatically complete the checkout.
+
+    If any error is raised during the process, it will be caught and logged.
+    """
+    checkout_id = graphene.Node.to_global_id("Checkout", checkout_pk)
+    try:
+        checkout = Checkout.objects.get(pk=checkout_pk)
+    except Checkout.DoesNotExist:
+        return
+
+    user = User.objects.filter(pk=user_id).first()
+    app = App.objects.filter(pk=app_id).first()
+    manager = get_plugins_manager(allow_replica=False)
+    lines, _ = fetch_checkout_lines(checkout)
+    checkout_info = fetch_checkout_info(checkout, lines, manager)
+
+    task_logger.info(
+        "Automatic checkout completion triggered for checkout: %s.",
+        checkout_id,
+        extra={"checkout_id": checkout_id},
+    )
+    try:
+        complete_checkout(
+            manager=manager,
+            checkout_info=checkout_info,
+            lines=lines,
+            payment_data={},
+            store_source=False,
+            user=user,
+            app=app,
+        )
+    except ValidationError as error:
+        task_logger.warning(
+            "Automatic checkout completion failed for checkout: %s.",
+            checkout_id,
+            extra={
+                "checkout_id": checkout_id,
+                "error": str(error),
+            },
+        )
+    else:
+        task_logger.info(
+            "Automatic checkout completion succeeded for checkout: %s.",
+            checkout_id,
+            extra={"checkout_id": checkout_id},
+        )
```

Comment:
Why do you use `else` here, instead of logging the success directly under `try` statement? If any other error than `ValidationError` occur during `complete_checkout` we will log success.

Reply:
In a `try` block, you should only include code that might raise an exception and needs to be handled. The goal is to keep the try block as minimal as possible, containing only the code that you expect might fail. That's why I logging in the `else` block